pilot/cmd/pilot-agent/app/wait.go

```diff
@@ -79,6 +86,5 @@ func checkIfReady(client *http.Client, url string) error {
 func init() {
 	waitCmd.PersistentFlags().IntVar(&timeoutSeconds, "timeoutSeconds", 60, "maximum number of seconds to wait for Envoy to be ready")
 	waitCmd.PersistentFlags().IntVar(&requestTimeoutMillis, "requestTimeoutMillis", 500, "number of milliseconds to wait for response")
-	waitCmd.PersistentFlags().IntVar(&periodMillis, "periodMillis", 500, "number of milliseconds to wait between attempts")
```

Comment:
do we need to add a deprecation warning here rather than removing?  I'm not sure how many people are using this but we don't want to pull the rug out from anyone.

Also since a CLI flag is getting removed, I think a release note should be added so it gets called out in the release/upgrade notes

Reply:
Similar comment, but I don't get why we are removing or deprecating at all. We can have a poll period and still exit early during sleep. Just do

```go
timeout := time.After(timeoutSeconds)
for {
 select{
  case <-timeout: ...
  case <- time.After(periodMillis): ...
}
```

That being said, I would support a PR to deprecate these and replace them with `period` and `requestTimeout` (without Millis... we can use DurationVar)

Reply:
> do we need to add a deprecation warning here rather than removing? I'm not sure how many people are using this but we don't want to pull the rug out from anyone.
> 
> Also since a CLI flag is getting removed, I think a release note should be added so it gets called out in the release/upgrade notes

It's a  good idea, I will add a deprecation warning and release note, thanks @GregHanson !

Reply:
> Similar comment, but I don't get why we are removing or deprecating at all. We can have a poll period and still exit early during sleep. Just do
> 
> ```go
> timeout := time.After(timeoutSeconds)
> for {
>  select{
>   case <-timeout: ...
>   case <- time.After(periodMillis): ...
> }
> ```
> 
> That being said, I would support a PR to deprecate these and replace them with `period` and `requestTimeout` (without Millis... we can use DurationVar)

Hi @howardjohn, if the checking logic is filled in the second case(`case <- time.After(periodMillis)`), there may be a timeout notification earlier than time.After notification, in this situation it would not be the really timeout, and it's the same as before, make sense?

Reply:
no I don't get it.

Reply:
I cannot get what bug it is fixing or refactor?

Reply:
Although the reproduce was not stable and very rare, I encountered this rare situation when doing verifications, the checking client showed timeout, but the gRPC server created the UDS file. 
So I analyze that this may be caused by the `sleep`. Let's assume that there may be 450 milliseconds left to be timeout, but the checking UDS file is still not ready, so the checking process will go to sleep 500 milliseconds. However, the timeout will be triggered once the sleep is over.

Reply:
Hi @howardjohn & @hzxuzhonghu, described as above

Reply:
That's why I recommend using a channel then you can cancel during the poll sleep

Reply:
Yeah, although `select` will randomly pick the case switch, `timeout` and `sleep` channel, I agree that use a channel would be a better method.

Reply:
> do we need to add a deprecation warning here rather than removing? I'm not sure how many people are using this but we don't want to pull the rug out from anyone.
> 
> Also since a CLI flag is getting removed, I think a release note should be added so it gets called out in the release/upgrade notes

Hi @GregHanson, there are some discussion with john and zhonghu, `periodMillis` flag may need to be changed instead of removing. Hope you know this, thanks for your good suggestion before!